### ISSUE TITLE
Degraded accuracy on Pascal VOC with mmdetection 2.0

### ISSUE THREAD

**Issue by TonyLianLong** [open-mmlab/mmdetection#2861]
Thanks for your error report and we appreciate it a lot.

**Checklist**
1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**
The mAP is 3% lower for mmdetection 2.0 than 1.x.

**Reproduction**
1. Use official method to train Pascal VOC with 2 GPUs. The result is different beginning from 1st epoch, so the data below are from the 1st epoch.
```
+-------------+------+-------+--------+-------+                                                                                                                                        
| class       | gts  | dets  | recall | ap    |                                                                                                                                        
+-------------+------+-------+--------+-------+                                                                                                                                        
| aeroplane   | 285  | 2272  | 0.912  | 0.731 |                                                                                                                                        
| bicycle     | 337  | 2191  | 0.917  | 0.753 |                                                                                                                                        
| bird        | 459  | 4582  | 0.924  | 0.743 |                                                                                                                                        
| boat        | 263  | 2724  | 0.859  | 0.557 |                                                                                                                                        
| bottle      | 469  | 4328  | 0.840  | 0.601 |                                                                                                                                        
| bus         | 213  | 2763  | 0.967  | 0.721 |                                                                                                                                        
| car         | 1201 | 7205  | 0.954  | 0.835 |                                                                                                                                        
| cat         | 358  | 2555  | 0.947  | 0.735 |                                                                                                                                        
| chair       | 756  | 8198  | 0.852  | 0.540 |                                                                                                                                        
| cow         | 244  | 1365  | 0.902  | 0.632 |                                                                                                                                        
| diningtable | 206  | 2842  | 0.864  | 0.555 |                                                                                                                                        
| dog         | 489  | 1912  | 0.914  | 0.661 |                                                                                                                                        
| horse       | 348  | 1714  | 0.911  | 0.701 |                                                                                                                                        
| motorbike   | 325  | 1473  | 0.914  | 0.759 |                                                                                                                                        
| person      | 4528 | 26464 | 0.949  | 0.802 |                                                                                                                                        
| pottedplant | 480  | 5522  | 0.806  | 0.434 |                                                                                                                                        
| sheep       | 242  | 1646  | 0.864  | 0.620 |                                                                                                                                        
| sofa        | 239  | 1775  | 0.929  | 0.657 |                                                                                                                                        
| train       | 282  | 2129  | 0.929  | 0.696 |                                                                                                                                        
| tvmonitor   | 308  | 3665  | 0.893  | 0.693 |                                                                                                                                        
+-------------+------+-------+--------+-------+                                                                                                                                        
| mAP         |      |       |        | 0.671 |                                                                                                                                        
+-------------+------+-------+--------+-------+          
```                                                                                                                             
2020-05-31 12:10:14,773 - mmdet - INFO - Epoch [1][3104/3104]   lr: 0.01000, mAP: 0.6714 
2. Use a 1.x version to train:
```
+-------------+------+-------+--------+-------+
| class       | gts  | dets  | recall | ap    |
+-------------+------+-------+--------+-------+
| aeroplane   | 285  | 1383  | 0.905  | 0.752 |
| bicycle     | 337  | 1383  | 0.920  | 0.788 |
| bird        | 459  | 2200  | 0.893  | 0.730 |
| boat        | 263  | 1785  | 0.863  | 0.600 |
| bottle      | 469  | 3773  | 0.846  | 0.646 |
| bus         | 213  | 947   | 0.915  | 0.747 |
| car         | 1201 | 6046  | 0.944  | 0.826 |
| cat         | 358  | 1812  | 0.950  | 0.839 |
| chair       | 756  | 5038  | 0.811  | 0.530 |
| cow         | 244  | 1629  | 0.955  | 0.676 |
| diningtable | 206  | 1985  | 0.893  | 0.578 |
| dog         | 489  | 3548  | 0.980  | 0.785 |
| horse       | 348  | 3133  | 0.971  | 0.778 |
| motorbike   | 325  | 1665  | 0.917  | 0.743 |
| person      | 4528 | 28480 | 0.961  | 0.811 |
| pottedplant | 480  | 3483  | 0.773  | 0.408 |
| sheep       | 242  | 1676  | 0.921  | 0.705 |
| sofa        | 239  | 4238  | 0.962  | 0.683 |
| train       | 282  | 1844  | 0.922  | 0.728 |
| tvmonitor   | 308  | 2721  | 0.883  | 0.709 |
+-------------+------+-------+--------+-------+
| mAP         |      |       |        | 0.703 |
+-------------+------+-------+--------+-------+
```
2020-02-26 14:08:06,568 - mmdet - INFO - Epoch [1][12414/12414] lr: 0.00250, mAP: 0.7031

**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment infomation and paste it here.
```
sys.platform: linux
Python: 3.7.7 (default, May  7 2020, 21:25:33) [GCC 7.3.0]
CUDA available: True
CUDA_HOME: /my/home/directory/cuda-10.1
NVCC: Cuda compilation tools, release 10.1, V10.1.243
GPU 0,1,2,3: GeForce GTX 1080 Ti
GCC: gcc (Debian 6.3.0-18+deb9u1) 6.3.0 20170516
PyTorch: 1.4.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - Intel(R) Math Kernel Library Version 2020.0.1 Product Build 20200208 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.1
  - Build settings: BLAS=MKL, BUILD_NAMEDTENSOR=OFF, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -O2 -fPIC -Wno-narrowing -Wall -Wextra -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Wno-stringop-overflow, DISABLE_NUMA=1, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF,

TorchVision: 0.5.0
OpenCV: 4.2.0
MMCV: 0.5.8
MMDetection: 1.2.0+0f33c08
MMDetection Compiler: GCC 7.3
MMDetection CUDA Compiler: 10.1
```
2. I've tried on different environments (including different hosts with multiple types of GPUs), but it is the same.
3. Once I switched to a 1.x version, the mAP improves.

**Comment by TonyLianLong** [comment]
What's more, we need a list of accuracies on Pascal VOC. There is no accuracies for Pascal VOC with mmdetection 2.0.

**Comment by ZwwWayne** [comment]
Hi @TonyLianLong ,
It seems that the batch sizes are different in these experiments, what are their final accuracies?

**Comment by TonyLianLong** [comment]
I currently do not have final accuracies in my hand, but I expect the 1.x version to have 3% higher mAP than the 2.0. I scaled learning rate linearly with batch sizes and it should not cause this degrade.

**Comment by TonyLianLong** [comment]
This could be reproduced as I tried both 1.x and 2.0 multiple times.

**Comment by ZwwWayne** [comment]
Which config are you using and what command you use?
Using `configs/pascal_voc/faster_rcnn_r50_fpn_1x_voc0712.py`, 4 runs obtain  0.8086, 0.7974, 0.7985, and 0.8009 AP, respectively.

**Comment by TonyLianLong** [comment]
I'm using the default config with distributed training on 2 GPUs, and the final mAP is around 0.803 on 1.x and 3% lower on 2.0. Are the mAP that you reported from 2.0?

**Comment by ZwwWayne** [comment]
Yes, the mAP is reported from 2.0 and trained in the past few days.

**Comment by hellock** [comment]
@TonyLianLong Check your learning rate if you were using only 2 GPUs.

**Comment by TonyLianLong** [comment]
If it does work on your end, then it seems like it's the issue with 2.0 on my end. However, 1.x works fine. How many GPUs do you use? @ZwwWayne 

> @TonyLianLong Check your learning rate if you were using only 2 GPUs.
I used linear scaling and use 0.25 * LR.

**Comment by TonyLianLong** [comment]
> @TonyLianLong Check your learning rate if you were using only 2 GPUs.
I used linear scaling and use 0.25 * LR.

**Comment by ZwwWayne** [comment]
> If it does work on your end, then it seems like it's the issue with 2.0 on my end. However, 1.x works fine. I'll close this issue and investigate into it. By the way, how many GPUs do you use? @ZwwWayne

I use 8GPUs to train the model. Usually for a small dataset using small batch size to train a model could result in unstable performance.

**Comment by fityanul** [comment]
> If it does work on your end, then it seems like it's the issue with 2.0 on my end. However, 1.x works fine. I'll close this issue and investigate into it. By the way, how many GPUs do you use? @ZwwWayne

Hi @TonyLianLong, have You solve this problem? I also faced the same problem but in the custom dataset with pascal voc label, 1.x works well than 2.0.
